test/integration/scheduler/podsready/scheduler_test.go

```diff
@@ -355,3 +355,97 @@ var _ = ginkgo.Describe("SchedulerWithWaitForPodsReady", func() {
 		})
 	})
 })
+
+var _ = ginkgo.Describe("SchedulerWithWaitForPodsReadyNonblockingMode", func() {
+
+	var (
+		defaultFlavor    *kueue.ResourceFlavor
+		podsReadyTimeout time.Duration
+		ns               *corev1.Namespace
+		prodClusterQ     *kueue.ClusterQueue
+		devClusterQ      *kueue.ClusterQueue
+		prodQueue        *kueue.LocalQueue
+		devQueue         *kueue.LocalQueue
+	)
+
+	ginkgo.JustBeforeEach(func() {
+		fwk = &framework.Framework{
+			ManagerSetup: func(mgr manager.Manager, ctx context.Context) {
+				managerAndSchedulerSetupWithTimeoutAdmission(mgr, ctx, podsReadyTimeout, false)
+			},
+			CRDPath:     filepath.Join("..", "..", "..", "..", "config", "components", "crd", "bases"),
+			WebhookPath: filepath.Join("..", "..", "..", "..", "config", "components", "webhook"),
+		}
+		ctx, cfg, k8sClient = fwk.Setup()
+
+		defaultFlavor = utiltesting.MakeResourceFlavor("default").Obj()
+		gomega.Expect(k8sClient.Create(ctx, defaultFlavor)).To(gomega.Succeed())
+
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "podsready-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+
+		prodClusterQ = testing.MakeClusterQueue("prod-cq").
+			Cohort("all").
+			ResourceGroup(*testing.MakeFlavorQuotas("default").Resource(corev1.ResourceCPU, "5").Obj()).
+			Obj()
+		gomega.Expect(k8sClient.Create(ctx, prodClusterQ)).Should(gomega.Succeed())
+
+		devClusterQ = testing.MakeClusterQueue("dev-cq").
+			Cohort("all").
+			ResourceGroup(*testing.MakeFlavorQuotas("default").Resource(corev1.ResourceCPU, "5").Obj()).
+			Obj()
+		gomega.Expect(k8sClient.Create(ctx, devClusterQ)).Should(gomega.Succeed())
+
+		prodQueue = testing.MakeLocalQueue("prod-queue", ns.Name).ClusterQueue(prodClusterQ.Name).Obj()
+		gomega.Expect(k8sClient.Create(ctx, prodQueue)).Should(gomega.Succeed())
+
+		devQueue = testing.MakeLocalQueue("dev-queue", ns.Name).ClusterQueue(devClusterQ.Name).Obj()
+		gomega.Expect(k8sClient.Create(ctx, devQueue)).Should(gomega.Succeed())
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, prodClusterQ, true)
+		util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, devClusterQ, true)
+		fwk.Teardown()
+	})
+
+	ginkgo.Context("Long PodsReady timeout", func() {
+
+		ginkgo.BeforeEach(func() {
+			podsReadyTimeout = time.Minute
+		})
+
+		ginkgo.It("Should not block admission of one new workload if two are considered in the same scheduling cycle", func() {
+			ginkgo.By("creating two workloads but delaying cluster queue creation which has enough capacity")
+			prodWl := testing.MakeWorkload("prod-wl", ns.Name).Queue(prodQueue.Name).Request(corev1.ResourceCPU, "11").Obj()
+			gomega.Expect(k8sClient.Create(ctx, prodWl)).Should(gomega.Succeed())
+			// wait a second to ensure the CreationTimestamps differ and scheduler picks the first created to be admitted
+			time.Sleep(time.Second)
```

Comment:
It does not seem needed here.

Reply:
Testing block admission needs 2 tasks, we need a not ready task to ensure the second task not been blocked

Reply:
yes, creating two tasks (or jobs) is fine, but I think we don't need to sleep to ensure they have different creation timestamps. The sleep was added in the original test to have control over which Job is admitted, so that the asserts are deterministic. Here you get both of them admitted, so the asserts don't make use of which one is admitted first.

Reply:
You are right, I will remove the sleep action